azurestack/resource_arm_loadbalancer_probe.go

```diff
@@ -0,0 +1,281 @@
+	d.Set("port", config.ProbePropertiesFormat.Port)
+	d.Set("request_path", config.ProbePropertiesFormat.RequestPath)
+
+	var load_balancer_rules []string
+	if config.ProbePropertiesFormat.LoadBalancingRules != nil {
+		for _, ruleConfig := range *config.ProbePropertiesFormat.LoadBalancingRules {
+			load_balancer_rules = append(load_balancer_rules, *ruleConfig.ID)
+		}
+	}
+	d.Set("load_balancer_rules", load_balancer_rules)
+
+	return nil
+}
+
+func resourceArmLoadBalancerProbeDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).loadBalancerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	loadBalancerID := d.Get("loadbalancer_id").(string)
+	armMutexKV.Lock(loadBalancerID)
+	defer armMutexKV.Unlock(loadBalancerID)
+
+	loadBalancer, exists, err := retrieveLoadBalancerById(loadBalancerID, meta)
+	if err != nil {
+		return fmt.Errorf("Error Getting LoadBalancer By ID: %+v", err)
+	}
+	if !exists {
+		d.SetId("")
+		return nil
+	}
+
+	_, index, exists := findLoadBalancerProbeByName(loadBalancer, d.Get("name").(string))
+	if !exists {
+		return nil
+	}
+
+	oldProbes := *loadBalancer.LoadBalancerPropertiesFormat.Probes
+	newProbes := append(oldProbes[:index], oldProbes[index+1:]...)
+	loadBalancer.LoadBalancerPropertiesFormat.Probes = &newProbes
+
+	resGroup, loadBalancerName, err := resourceGroupAndLBNameFromId(d.Get("loadbalancer_id").(string))
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer Name and Group: {{err}}", err)
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resGroup, loadBalancerName, *loadBalancer)
+	if err != nil {
+		return fmt.Errorf("Error Creating/Updating LoadBalancer %q (Resource Group %q): %+v", loadBalancerName, resGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for completion of LoadBalancer %q (Resource Group %q): %+v", loadBalancerName, resGroup, err)
+	}
+
+	read, err := client.Get(ctx, resGroup, loadBalancerName, "")
+	if err != nil {
+		return errwrap.Wrapf("Error Getting LoadBalancer {{err}}", err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read LoadBalancer %s (resource group %s) ID", loadBalancerName, resGroup)
+	}
+
+	return nil
+}
+
+func expandAzureRmLoadBalancerProbe(d *schema.ResourceData, lb *network.LoadBalancer) (*network.Probe, error) {
+
+	properties := network.ProbePropertiesFormat{
+		NumberOfProbes:    utils.Int32(int32(d.Get("number_of_probes").(int))),
+		IntervalInSeconds: utils.Int32(int32(d.Get("interval_in_seconds").(int))),
+		Port:              utils.Int32(int32(d.Get("port").(int))),
+	}
+
+	if v, ok := d.GetOk("protocol"); ok {
+		properties.Protocol = network.ProbeProtocol(v.(string))
+	}
+
+	if v, ok := d.GetOk("request_path"); ok {
+		properties.RequestPath = utils.String(v.(string))
+	}
+
+	probe := network.Probe{
+		Name: utils.String(d.Get("name").(string)),
+		ProbePropertiesFormat: &properties,
+	}
+
+	return &probe, nil
```

Comment:
This could be written as:
```go
return &network.Probe{
		Name: utils.String(d.Get("name").(string)),
		ProbePropertiesFormat: &properties,
	}
```